File: cpp/src/WritingUtensil.cpp
#include "WritingUtensil.h"

#include <iostream>

namespace marker {
   Color WritingUtensil::getColor() const {
       return color;
   }

   void WritingUtensil::setColor(Color _color) {
       color = _color;
   }

   WritingUtensil::WritingUtensil(Color _color)
    : color(_color) 
    {
        std::cout << "made " << color << " wu at " << (void*) (this) << std::endl;
    }
    
    WritingUtensil::~WritingUtensil() {
        
    }
}

File: java/marker/src/main/java/com/github/wmacevoy/marker/WritingUtensil.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.github.wmacevoy.marker;

/**
 *
 * @author wmacevoy
 */
public interface WritingUtensil {
    static final Color DEFAULT_COLOR = Color.BLACK;
    Color getColor(); 
    void setColor(Color color);
    void write() throws BadWriteException;
}
